tools/formatting.eslintrc.yaml

```diff
@@ -2,10 +2,13 @@
 # we can follow each Prettier run with a streamlined (and faster) ESLint run
 # that uses only these rules.
 
-parser: babel-eslint
+parser: "@babel/eslint-parser"
 
 plugins:
   - flowtype
+  - import
+  - jest
+  - react
```

Comment:
Cool. The one downside of a change like this is that it makes a formatting run slower, by giving it more code to load. But I just tested, and the difference is tolerable:

Before:
```
$ echo >>src/ZulipMobile.js ; time tools/fmt 
success formatting 1 file with prettier-eslint
time: 3.988s wall (4.999s u, 0.240s s)

$ echo >>src/ZulipMobile.js ; time tools/fmt 
success formatting 1 file with prettier-eslint
time: 4.022s wall (5.130s u, 0.220s s)
```

After:
```
$ echo >>src/ZulipMobile.js ; time tools/fmt 
success formatting 1 file with prettier-eslint
time: 4.275s wall (5.452s u, 0.288s s)

$ echo >>src/ZulipMobile.js ; time tools/fmt 
success formatting 1 file with prettier-eslint
time: 4.202s wall (5.303s u, 0.229s s)
```

So like 200-300ms longer, which is enough time to be noticeable… except that it's much more annoyingly slow already at 4.0s.

So increasing that by 5-7%, to 4.2s or 4.3s, is worth it to get rid of those spurious "Definition for rule '…' was not found" errors.

Reply:
> The one downside of a change like this is that it makes a formatting run slower, by giving it more code to load. But I just tested, and the difference is tolerable […]

Thanks for running the timer! If it's loading more code, I guess I'm not sure how the resulting config ends up being so similar with and without these added plugins: https://github.com/zulip/zulip-mobile/pull/5393#issuecomment-1146275599

Reply:
> Thanks for running the timer! If these three lines are causing more code to be loaded, I guess I'm not sure how the resulting config ends up being so similar with and without the lines: [#5393 (comment)](https://github.com/zulip/zulip-mobile/pull/5393#issuecomment-1146275599)

Ah. Here's that command:
> `yarn eslint --print-config tools/formatting.eslintrc.yaml`

But that's not doing what you want it to be doing. That's saying: what is the ESLint config *for linting the file `tools/formatting.eslintrc.yaml`*. IOW, `--print-config` doesn't take an argument.

Here's notes from the top of our eslintrc:
```
# Tips for developing on this file:
#
#  * To see what config currently applies, try a command like:
#
#      $ npx eslint --print-config src/types.js | less
#
#    or for a specific rule:
#
#      $ npx eslint --print-config src/types.js | jq '.rules["no-continue"]'
```

And here's the `eslint` command line from `tools/fmt`:
```
eslint --no-eslintrc -c tools/formatting.eslintrc.yaml --fix "${files[@]}"
```

So, with our regular ESLint config:
```
$ node_modules/.bin/eslint --print-config src/types.js | jq .plugins
[
  "jsx-a11y",
  "react-hooks",
  "react",
  "import",
  "flowtype",
  "jest"
]
```

With the formatting-only config, at the tip of this branch:
```
$ node_modules/.bin/eslint --no-eslintrc -c tools/formatting.eslintrc.yaml --print-config src/types.js | jq .plugins
[
  "react",
  "jest",
  "import",
  "flowtype"
]
```

With the formatting-only config, before this branch:
```
$ node_modules/.bin/eslint --no-eslintrc -c tools/formatting.eslintrc.yaml --print-config src/types.js | jq .plugins
[
  "flowtype"
]
```